src/Api/State/SearchBeneficiaryProvider.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+namespace App\Api\State;
+
+use ApiPlatform\Metadata\Operation;
+use ApiPlatform\State\ProviderInterface;
+use App\Repository\UserRepository;
+
+readonly class SearchBeneficiaryProvider implements ProviderInterface
+{
+    public function __construct(
+        private readonly UserRepository $userRepository,
+    ) {
+    }
+
+    public function provide(Operation $operation, array $uriVariables = [], array $context = []): object|array|null
+    {
+        if (!$context['filters']) {
+            return null;
+        }
+        $user = $this->userRepository->findByUsername(trim($context['filters']['username'], '"') ?? '');
```

Comment:
at this point, `$context['filters']['username']` could not exist right ?

Reply:
Ah yes, i added trim later, it may not work this way. Thx